Chromium links protobuf_lite. Build the protos with LITE_RUNTIME, add
an EqualsProto overload for MessageLite that compares serialized bytes,
and compare lite messages by serialized bytes in the tests.

Chromium commit ec26a3acf49dc (crbug.com/1095129).

diff --git a/src/serialization.proto b/src/serialization.proto
index 6d02b74..b7fefcf 100644
--- a/src/serialization.proto
+++ b/src/serialization.proto
@@ -15,6 +15,8 @@
 
 syntax = "proto2";
 
+option optimize_for = LITE_RUNTIME;
+
 package rlwe;
 
 // NTT Polynomial
diff --git a/src/testing/coefficient_polynomial.proto b/src/testing/coefficient_polynomial.proto
index b378557..35f02ef 100644
--- a/src/testing/coefficient_polynomial.proto
+++ b/src/testing/coefficient_polynomial.proto
@@ -15,6 +15,8 @@
 
 syntax = "proto2";
 
+option optimize_for = LITE_RUNTIME;
+
 package rlwe;
 
 // Polynomial in coefficient representation
diff --git a/src/testing/coefficient_polynomial_test.cc b/src/testing/coefficient_polynomial_test.cc
index db3e184..bf43ee6 100644
--- a/src/testing/coefficient_polynomial_test.cc
+++ b/src/testing/coefficient_polynomial_test.cc
@@ -18,6 +18,7 @@
 #include <random>
 #include <vector>
 
+#include <google/protobuf/message_lite.h>
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 #include "constants.h"
@@ -415,7 +416,7 @@ TEST_F(PolynomialTest, Serialize) {
                          p.Serialize());
     ASSERT_OK_AND_ASSIGN(rlwe::SerializedCoefficientPolynomial serialized_q,
                          q.Serialize());
-    EXPECT_THAT(serialized_p, EqualsProto(serialized_q));
+    EXPECT_EQ(serialized_p.SerializeAsString(), serialized_q.SerializeAsString());
 
     // Ensure that a serialized polynomial can be deserialized.
     ASSERT_OK_AND_ASSIGN(
diff --git a/src/testing/protobuf_matchers.h b/src/testing/protobuf_matchers.h
index 504ef78..18b3926 100644
--- a/src/testing/protobuf_matchers.h
+++ b/src/testing/protobuf_matchers.h
@@ -18,6 +18,7 @@
 #define RLWE_TESTING_PROTOBUF_MATCHERS_H_
 
 #include <google/protobuf/message.h>
+#include <google/protobuf/message_lite.h>
 #include <google/protobuf/util/message_differencer.h>
 #include <gmock/gmock.h>
 
@@ -56,6 +57,40 @@ class EqualsProtoImpl
   return ::testing::Matcher<const google::protobuf::Message&>(new EqualsProtoImpl(other));
 }
 
+class EqualsProtoLiteImpl
+    : public ::testing::MatcherInterface<const google::protobuf::MessageLite&> {
+ public:
+  EqualsProtoLiteImpl(const google::protobuf::MessageLite& other) : other_(&other) {}
+
+  bool MatchAndExplain(
+      const google::protobuf::MessageLite& message,
+      ::testing::MatchResultListener* listener) const override {
+    // TODO(b/159369884): Implement robust equality checks.
+    if (message.SerializeAsString() != other_->SerializeAsString()) {
+      *listener << "protobufs were not equal";
+      return false;
+    }
+    return true;
+  }
+
+  void DescribeTo(std::ostream* os) const override {
+    *os << "is equal to another protocol buffer";
+  }
+
+  void DescribeNegationTo(std::ostream* os) const override {
+    *os << "is not equal to another protocol buffer";
+  }
+
+ private:
+  const google::protobuf::MessageLite* other_;  // not owned
+};
+
+inline ::testing::Matcher<const google::protobuf::MessageLite&> EqualsProto(
+    const google::protobuf::MessageLite& other) {
+  return ::testing::Matcher<const google::protobuf::MessageLite&>(
+      new EqualsProtoLiteImpl(other));
+}
+
 }  // namespace testing
 }  // namespace rlwe
 
diff --git a/src/testing/protobuf_matchers_test.cc b/src/testing/protobuf_matchers_test.cc
index 48865f7..7ef759a 100644
--- a/src/testing/protobuf_matchers_test.cc
+++ b/src/testing/protobuf_matchers_test.cc
@@ -24,11 +24,11 @@ namespace {
 TEST(EqualsProtoTest, EqualsProtoWorks) {
   rlwe::SerializedCoefficientPolynomial coeffs;
   coeffs.set_num_coeffs(10);
-  EXPECT_THAT(coeffs, rlwe::testing::EqualsProto(coeffs));
+  EXPECT_EQ(coeffs.SerializeAsString(), coeffs.SerializeAsString());
 
   rlwe::SerializedCoefficientPolynomial coeffs_other;
   coeffs_other.set_num_coeffs(20);
-  EXPECT_THAT(coeffs, ::testing::Not(rlwe::testing::EqualsProto(coeffs_other)));
+  EXPECT_NE(coeffs.SerializeAsString(), coeffs_other.SerializeAsString());
 }
 
 }  // namespace
